bittide/src/Bittide/Arithmetic/Time.hs

```diff
@@ -71,3 +79,35 @@ trueFor _ clk rst =
   transF counter = \case
     True -> satSucc SatBound counter
     _    -> 0
+
+-- | Rises after the incoming signal has been 'True' for the specified amount of
+-- time. The time is specified in milliseconds. Example invocation:
+--
+-- > trueForSteps @(Milliseconds 1) Proxy someLimit clk rst signal
+--
+-- Use 'trueFor' if you know the time to wait for at compile time. If not, use
+-- 'trueForSteps'.
+trueForSteps ::
+  forall (stepSize :: Nat) (dom :: Domain) (a :: Type) .
+  ( HasCallStack
+  , KnownDomain dom
+  , KnownNat stepSize
+  , NFDataX a, Bounded a, Counter a, Eq a, Num a
+  ) =>
+  Proxy stepSize ->
+  -- ^ Step size. Use the type aliases of 'Bittide.Arithmetic.Time' for time span
+  -- specification.
+  a ->
```

Comment:
Why not naming the argument `counter` here. There is no need for first letting the user find that this is a counter because it's constrained by the `Counter` class.

Reply:
Agreed